treeherder/perf/migrations/0018_remove_old_frameworks.py

```diff
@@ -0,0 +1,21 @@
+# Generated by Django 2.2.4 on 2019-09-17 12:23
+
+from django.db import migrations
+
+
+def delete_old_frameworks(apps, schema_editor):
+    PerformanceFramework = apps.get_model('perf', 'PerformanceFramework')
+    frameworks_to_remove = ['talos-aws', 'hasal', 'servo-perf', 'autophone']
+
+    PerformanceFramework.objects.filter(name__in=frameworks_to_remove).delete()
+
+
+class Migration(migrations.Migration):
+
+    dependencies = [
+        ('perf', '0017_assignment_support_for_alert_summaries'),
+    ]
+
+    operations = [
+        migrations.RunPython(delete_old_frameworks, migrations.RunPython.noop)
```

Comment:
Is this actually needed? Since the these fields have been removed from the fixtures, wouldn't that mean that the next time migrations are run on stage and production that these frameworks simply wouldn't exist anymore?

In general, I don't think this is the best approach to deleting old data. I think it'd be better to either set up a Django command with (for example) frameworks passed in as an arg if there might be a need to periodically delete certain data or for someone on the TH team to manually run the SQL equivalent to delete certain data against the RDS instances.

Reply:
I tested with the `loaddata` command, in order to see if loading the data while old frameworks removed from fixtures would remove the old frameworks from DB.
**The steps:**
    **1.** run the 0017_assignment_support_for_alert_summaries migration (invalidate the newly-added 0018_remove_old_frameworks.py)
    **2.** remove old frameworks ('talos-aws', 'hasal', 'servo-perf', 'autophone') from performance_framework.json fixture
    **3.** run a ./manage.py load_initial_data
    **4.** the old frameworks are still available in the DB
    
I found here: `https://stackoverflow.com/questions/35089887/django-loaddata-to-update-data`
that the fixtures are meant to provide initial data. They alone cannot remove existing data.

**Conlusion**:
    There are two ways to solve this problem:
        -migration
        -Django command

Reply:
> Is this actually needed? Since the these fields have been removed from the fixtures, wouldn't that mean that the next time migrations are run on stage and production that these frameworks simply wouldn't exist anymore?

I find the official docs ambiguous on this matter.
> Each time you run loaddata, the data will be read from the fixture and re-loaded into the database. Note this means that if you change one of the rows created by a fixture and then run loaddata again, you’ll wipe out any changes you’ve made.

I attempted a `./manage.py loaddata performance_framework.json performance_bug_template.yaml` on my local machine, with those frameworks removed. Thing is it didn't delete them. It didn't even throw a warning or error.
 
> In general, I don't think this is the best approach to deleting old data. I think it'd be better to either set up a Django command with (for example) frameworks passed in as an arg if there might be a need to periodically delete certain data or for someone on the TH team to manually run the SQL equivalent to delete certain data against the RDS instances.

We found a situation similar to ours and followed the advice [there](https://stackoverflow.com/questions/35089887/django-loaddata-to-update-data). We liked this approach because is CI-agnostic, in that it will auto clean things on database. You don't need to be aware of which staging/production environments the app is running on.

Reply:
Will work on a dedicate management command.